core/src/main/java/org/keycloak/crypto/KeyWrapper.java

```diff
@@ -99,19 +99,19 @@ public void setSecretKey(SecretKey secretKey) {
     }
 
     public Key getSignKey() {
-        return signKey;
+        return privateKey;
```

Comment:
IMO, it might be good that KeyWrapper does not provide usage-dependent methods (namely, get/setSignKey(), get/setVerifyKey()).
On behalf of them, it provides algorithm-dependent method (namely, get/setPublicKey(), get/setPrivateKey(), get/setSecretKey()).

The reason are as follows:

1) current KeyWrapper assumes that keys of public key cryptography be used as signing/verifying. But this PR introduces new usage of such keys, namely, encrypting/decrypting.

2) On signing/verifying, current KeyWrapper provides  usage-dependent methods for public key cryptography while algorithm-dependent method for shared key cryptography. There is such the difference between algorithms of key.

It might be good that the role of  KeyWrapper is treating algorithms of keys while the role of users of KeyWrapper is treating their usage.  
(e.g. 
when encrypting in public key cryptography, the user of KeyWrapper calls KeyWrapper.getPublicKey().
when signing in public key cryptography, the user of KeyWrapper calls KeyWrapper.getPrivateKey().
when signing in shared key cryptography, the user of KeyWrapper calls KeyWrapper.getSecretKey().
)

Reply:
I completely agree with that. Just a note that in my comment above, I suggested to remove the get/setSignKey(), get/setVerifyKey() from the KeyWrapper class and replace the existing invocations of those methods in Keycloak codebase with the get/setPublicKey(), get/setPrivateKey() if possible. Do you think you can do it in this PR? Sorry if it was unclear in my previous comment.